src/wallet/walletdb.cpp

```diff
@@ -1144,15 +1147,16 @@ bool WalletBatch::EraseRecords(const std::unordered_set<std::string>& types)
     }
 
     // Iterate the DB and look for any records that have the type prefixes
-    while (true)
-    {
+    while (true) {
         // Read next record
         DataStream key{};
         DataStream value{};
         DatabaseCursor::Status status = cursor->Next(key, value);
         if (status == DatabaseCursor::Status::DONE) {
             break;
         } else if (status == DatabaseCursor::Status::FAIL) {
+            cursor.reset(nullptr);
```

Comment:
A note for myself: Seems unfortunate to `reset` a `std::unique_ptr` instance explicitly instead of using its RAII capabilities.

Reply:
Yeah, I explained the reason inside the commit description ca1871d48558b356257d2a95b046f63739dfe3a0:

> extra note from the Db.cursor doc:
"If transaction protection is enabled, cursors must be
opened and closed within the context of a transaction"
>
> thus why added a `CloseCursor` call before calling to
`TxnAbort/TxnCommit`.

The `CloseCursor` call is the ptr reset.
Just extra safety measures.